src/tasks/admin/delete-subscription.js

```diff
@@ -0,0 +1,44 @@
+const {success, failure} = require('../../lib/result');
+const taskError = require('../../lib/error/task');
+const createTaskError = require('../../lib/error/task');
+
+/**
+ * Init delete subscription task
+ * @param {RealmRepository} realmRepository Realm repository
+ * @param {SubscriptionRepository} subscriptionRepository Subscription repository
+ * @param {CommonTasks#sendSubscriptionSyncMessage} sendSubscriptionSyncMessage Task for sending subscription sync
+ * @returns {AdminTasks#deleteSubscription} Task
+ */
+module.exports = ({
+  realmRepository,
+  subscriptionRepository,
+  sendSubscriptionSyncMessage,
+}) =>
+  /**
+   * @function AdminTasks#deleteSubscription
+   * @param {string} realmId Realm id
+   * @param {string} id Subscription id
+   * @returns {Result<SubscriptionModel>}
+   */
+  async ({realmId, id}) => {
+    const realm = await realmRepository.findOne({id: realmId});
+    if (!realm) {
+      return failure(createTaskError({
+        code: 'UnknownRealm',
+        message: 'Cannot lookup the given realm.',
+      }));
+    }
+
+    const subscription = await subscriptionRepository.findOne({realmId, id});
+    if (!subscription) {
+      return failure(taskError({
+        code: 'UnknownSubscription',
+        message: 'Subscription does not exists.',
+      }));
+    }
+
+    await subscriptionRepository.deleteOne({realmId, id});
+    sendSubscriptionSyncMessage({subscription, action: 'deleted'});
```

Comment:
**thought:** Existing subscriptions within the broker are not covered by this. As long as a client holds a connection with an existing subscription open the client is able to receive messages on that channel.

Reply:
VerneMQ allows to [reevaluate existing subscriptions via command](https://docs.vernemq.com/live-administration/managing-sessions#managing-sessions). Something similar should be possible to be implemented as a broker side API endpoint or some kind message handler within `$SYS`.

Reply:
Let's address this in a dedicated PR

Reply:
https://github.com/realmq/realmq-platform/issues/148